SamarthTech/python-projects-2024
# Realtime Face Detection using OpenCV

# Real-Time Face Detection using OpenCV

#### Description:
This PR introduces a real-time face detection feature using OpenCV and Haar Cascade. The system captures live video feed from the webcam, detects faces in real-time, and highlights them with a rectangle.

#### Key Changes:
- Added `face_detection.py` script that captures video from the webcam.
- Integrated Haar Cascade (`haarcascade_frontalface_default.xml`) for face detection.
- Converts video frames to grayscale and detects faces.
- Draws a rectangle around detected faces in each frame.
- Program terminates when the 'q' key is pressed.

#### How to Test:
1. Run the script using `python face_detection.py`.
2. Ensure the webcam opens and faces are detected.
3. Press 'q' to exit the program.

This feature enhances the application with basic face recognition capabilities, providing a foundation for further development in computer vision.
